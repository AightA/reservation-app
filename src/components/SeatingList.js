import React from 'react';
import { connect } from 'react-redux';
import { Table } from 'react-bootstrap';
import { Input } from 'reactstrap';
import { useFormik } from 'formik';
import * as Yup from 'yup';

const SeatingList = ({ bookings }) => {
	const formik = useFormik({
		initialValues: {
			isArrived: false,
			seated: false,
		},

		validationSchema: Yup.object({
			isArrived: Yup.string().required('Required'),
			seated: Yup.string().required('Required'),
		}),
	});
	return (
		<div>
			<Table responsive="sm" striped bordered hover size="md">
				<thead>
					<tr>
						<th style={{ backgroundColor: 'grey' }}>#</th>
						<th style={{ backgroundColor: 'grey' }}>First name</th>
						<th style={{ backgroundColor: 'grey' }}>Last name</th>
						<th style={{ backgroundColor: 'grey' }}>Email</th>
						<th style={{ backgroundColor: 'grey' }}>Phone Number</th>
						<th style={{ backgroundColor: 'grey' }}>Guests</th>
						<th style={{ backgroundColor: 'grey' }}>Dining Date</th>
						<th style={{ backgroundColor: 'grey' }}>Arrived</th>
						<th style={{ backgroundColor: 'grey' }}>Seated</th>
					</tr>
				</thead>
				<tbody>
					{bookings.map((item, index) => {
						const {
							firstName,
							lastName,
							email,
							phoneNumber,
							numberOfPeople,
							diningDate,
						} = item;
						return (
							<tr key={index}>
								<td>{index + 1}</td>
								<td>{firstName}</td>
								<td>{lastName}</td>
								<td>{email}</td>
								<td>{phoneNumber}</td>
								<td>{numberOfPeople}</td>
								<td>{diningDate}</td>
								<td>
									<Input
										name="isArrived"
										id={index}
										type="checkbox"
										{...formik.getFieldProps('isArrived')}
									/>
									{formik.touched.isArrived && formik.errors.isArrived ? (
										<div>{formik.errors.isArrived}</div>
									) : null}
								</td>
								<td>
									<Input
										name="seated"
										id={index}
										type="checkbox"
										{...formik.getFieldProps('seated')}
									/>
									{formik.touched.seated && formik.errors.seated ? (
										<div>{formik.errors.seated}</div>
									) : null}
								</td>
							</tr>
						);
					})}
				</tbody>
			</Table>
		</div>
	);
};

const mapStateToProps = (state) => {
	return {
		bookings: state.bookings,
	};
};

export default connect(mapStateToProps)(SeatingList);
