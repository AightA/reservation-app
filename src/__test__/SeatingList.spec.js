import React from 'react';
import { render } from '../testUtils/helper';
import SeatingList from '../components/SeatingList';

describe('SeatingList snapshots', () => {
	it('Renders the connected app with initialState', () => {
		const result = render(<SeatingList />, {
			initialState: {
				bookings: [
					{
						id: 1,
						firstName: 'Mozafer',
					},
					{
						id: 2,
						firstName: 'Vicky',
					},
				],
			},
		});
		expect(result.getByText(/Mozafer/i)).toBeInTheDocument();
		expect(result.getByText(/vicky/i)).toBeInTheDocument();
		expect(result).toMatchSnapshot();
	});
});
