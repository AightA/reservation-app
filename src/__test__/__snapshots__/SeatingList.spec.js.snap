// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeatingList snapshots Renders the connected app with initialState 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="table-responsive-sm"
        >
          <table
            class="table table-md table-striped table-bordered table-hover"
          >
            <thead>
              <tr>
                <th
                  style="background-color: grey;"
                >
                  #
                </th>
                <th
                  style="background-color: grey;"
                >
                  First name
                </th>
                <th
                  style="background-color: grey;"
                >
                  Last name
                </th>
                <th
                  style="background-color: grey;"
                >
                  Email
                </th>
                <th
                  style="background-color: grey;"
                >
                  Phone Number
                </th>
                <th
                  style="background-color: grey;"
                >
                  Guests
                </th>
                <th
                  style="background-color: grey;"
                >
                  Dining Date
                </th>
                <th
                  style="background-color: grey;"
                >
                  Arrived
                </th>
                <th
                  style="background-color: grey;"
                >
                  Seated
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  1
                </td>
                <td>
                  Mozafer
                </td>
                <td />
                <td />
                <td />
                <td />
                <td />
                <td>
                  <input
                    class="form-check-input"
                    id="0"
                    name="isArrived"
                    type="checkbox"
                    value="false"
                  />
                </td>
                <td>
                  <input
                    class="form-check-input"
                    id="0"
                    name="seated"
                    type="checkbox"
                    value="false"
                  />
                </td>
              </tr>
              <tr>
                <td>
                  2
                </td>
                <td>
                  Vicky
                </td>
                <td />
                <td />
                <td />
                <td />
                <td />
                <td>
                  <input
                    class="form-check-input"
                    id="1"
                    name="isArrived"
                    type="checkbox"
                    value="false"
                  />
                </td>
                <td>
                  <input
                    class="form-check-input"
                    id="1"
                    name="seated"
                    type="checkbox"
                    value="false"
                  />
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="table-responsive-sm"
      >
        <table
          class="table table-md table-striped table-bordered table-hover"
        >
          <thead>
            <tr>
              <th
                style="background-color: grey;"
              >
                #
              </th>
              <th
                style="background-color: grey;"
              >
                First name
              </th>
              <th
                style="background-color: grey;"
              >
                Last name
              </th>
              <th
                style="background-color: grey;"
              >
                Email
              </th>
              <th
                style="background-color: grey;"
              >
                Phone Number
              </th>
              <th
                style="background-color: grey;"
              >
                Guests
              </th>
              <th
                style="background-color: grey;"
              >
                Dining Date
              </th>
              <th
                style="background-color: grey;"
              >
                Arrived
              </th>
              <th
                style="background-color: grey;"
              >
                Seated
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                1
              </td>
              <td>
                Mozafer
              </td>
              <td />
              <td />
              <td />
              <td />
              <td />
              <td>
                <input
                  class="form-check-input"
                  id="0"
                  name="isArrived"
                  type="checkbox"
                  value="false"
                />
              </td>
              <td>
                <input
                  class="form-check-input"
                  id="0"
                  name="seated"
                  type="checkbox"
                  value="false"
                />
              </td>
            </tr>
            <tr>
              <td>
                2
              </td>
              <td>
                Vicky
              </td>
              <td />
              <td />
              <td />
              <td />
              <td />
              <td>
                <input
                  class="form-check-input"
                  id="1"
                  name="isArrived"
                  type="checkbox"
                  value="false"
                />
              </td>
              <td>
                <input
                  class="form-check-input"
                  id="1"
                  name="seated"
                  type="checkbox"
                  value="false"
                />
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
